theano/sandbox/cuda/nvcc_compiler.py

```diff
@@ -214,11 +214,11 @@ def compile_str(
         if os.path.abspath(os.path.split(__file__)[0]) not in include_dirs:
             include_dirs.append(os.path.abspath(os.path.split(__file__)[0]))
 
-        libs = std_libs() + libs
+        libs = libs + std_libs()
         if 'cudart' not in libs:
             libs.append('cudart')
 
-        lib_dirs = std_lib_dirs() + lib_dirs
```

Comment:
I think the real solution is to remove empty string from the list of header dir and library dir list. So don't merge this PR now.